treeherder/perf/models.py

```diff
@@ -143,34 +131,35 @@ def cycle_data(self, repository, cycle_interval, chunk_size, sleep_time,
         # also remove any signatures which are (no longer) associated with
         # a job
         logger.warning('Removing performance signatures with missing jobs...')
-        for signature in PerformanceSignature.objects.filter(
-                repository=repository):
+        for signature in PerformanceSignature.objects.all():
+            if self._should_quit(started_at, max_overall_runtime):
+                logger.warning('Max runtime exceeded. Quitting cycling of performance signatures')
+                break
             if not self.filter(
-                    repository=repository,  # redundant, but leverages (repository, signature) compound index
-                    signature=signature).exists():
+                    repository_id=signature.repository_id,  # leverages (repository, signature) compound index
+                    signature_id=signature.id).exists():
                 signature.delete()
 
-    def _delete_in_chunks(self, repository, max_timestamp, chunk_size, sleep_time, logger,
+    def _delete_in_chunks(self, max_timestamp, max_chunk_size, logger,
                           started_at, max_overall_runtime):
         from django.db import connection
         any_succesful_attempt = False
 
         with connection.cursor() as cursor:
             while True:
                 if self._should_quit(started_at, max_overall_runtime):
-                    logger.info('Max runtime exceeded. Quitting cycling for repository {}'
-                                .format(repository.name))
+                    logger.warning('Max runtime exceeded. Quitting cycling of performance datum')
                     break
 
                 try:
+                    ideal_chunk_size = self._compute_ideal_chunk_size(max_timestamp, max_chunk_size)
                     cursor.execute('''
                         DELETE FROM `performance_datum`
-                        WHERE repository_id = %s AND
-                            push_timestamp < %s
+                        WHERE push_timestamp < %s
                         LIMIT %s
-                    ''', [repository.id, max_timestamp, chunk_size])
+                    ''', [max_timestamp, ideal_chunk_size])
```

Comment:
Could you you do something like this instead of the raw SQL? 
https://stackoverflow.com/questions/13248593/django-database-delete-specific-number-of-entries

Reply:
> Could you you do something like this instead of the raw SQL?
> https://stackoverflow.com/questions/13248593/django-database-delete-specific-number-of-entries

I tried to do that locally, but I get `django.db.utils.NotSupportedError: (1235, "This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'")`, just like one of the Stackoverflow users reported.

Reply:
What version do you need?
AWS allow us to upgrade to 8.0.15 and 5.7.26 (we're on 5.7.23)